### DISCUSSION TITLE
Python + Elasticsearch = Connection timeout

### DISCUSSION THREAD

**Discussion by ebdavison** [Security-Onion-Solutions/securityonion#13484] [General]
Following along here:
https://docs.securityonion.net/en/2.4/jupyter.html

Yet, I keep getting a connection timeout.

I have tried to use wrong credentials and I get an auth error.  But, when I run the same code with proper credentials, I get connection timeout so it does not make sense.

I did follow along and add my IP to the elasticsearch_rest config and can see my IP in the iptables output.

So, the network layer looks right.

Any tips here?  I can share code later today when I have more time if needed but basically I just copied from the above referenced article to try to get a basic query working.

**Comment by reyesj2** [comment]
Are you able to nc from the jupyter instance to elasticsearch? 
```nc -vz <securityonion-ip> 9200```
